examples/stm32f4_discovery/can_m2515/main.cpp

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2013, Kevin Läufer
+ * Copyright (c) 2013-2017, Niklas Hauser
+ * Copyright (c) 2016, Raphael Lehmann
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#include <modm/board.hpp>
+#include <modm/driver/can/mcp2515.hpp>
+#include <modm/processing/protothread.hpp>
+
+// If you use a different SPI instance, you may have to also choose different
+// GPIOs to connect to.
+using Int = GpioOutputB11;
+using Cs = GpioOutputB12;
+using Sck = GpioOutputB13;
+using Mosi = GpioOutputB15;
+using Miso = GpioInputB14;
+using SpiMaster = SpiMaster2;
+// Note that you can also use a bit-banged SPI driver as a drop-in replacement
+// using SpiMaster = BitBangSpiMaster<Sck, Mosi, Miso>;
+
+// Default filters to receive any extended CAN frame
+FLASH_STORAGE(uint8_t canFilter[]) = {
```

Comment:
I would also like to add a function that allows setting can filters without using this horrible old `FLASH_STORAGE` macros. (You don't have to.)

Reply:
well ill do it , i dont like it either

Reply:
I heavily dislike the way the filters are specified. They are passed as a raw `uint8_t[]` array defined by macros. When users get the number of filters or order of macros wrong it will write random values to the filter registers. We should do something more reasonable, like passing struct? Any other ideas?

Reply:
I would have done something about it but I don't understand what is happening there :D so I left it as is ... It was bad before .. it can be bad "for now"